charms/istio-pilot/src/charm.py

```diff
@@ -347,6 +346,31 @@ def _get_ingress_auth_data(self) -> dict:
         """
         raise NotImplementedError()
 
+    def _get_gateway_service(self):
+        """Returns a lightkube Service object for the gateway service."""
+        # FIXME: service name is configured via config, but it should really be provided directly
+        #  from the istio-gateway.  Providing here as a config at least makes this less rigid than
+        #  assuming the name.
+        # TODO: What happens if this service does not exist?  We should check on that and then add
+        #  tests to confirm this works
+        svc = self.lightkube_client.get(
+            Service, name=self.model.config["gateway-service-name"], namespace=self.model.name
+        )
+        return svc
+
+    def _is_gateway_service_up(self):
```

Comment:
maybe convert to property ?

Reply:
I'm never sure when to use a property.  Do you have a guideline to recommend?  

My rule of thumb is to do it when there's essentially no computation behind the item.  This one has very trivial computation behind it, but lots of logic across a few helpers.  So I'm torn :D

Reply:
I think the whole idea of properties to have computation behind :)
otherwise you use just class attribute

Reply:
done!